crates/secrets/src/lib.rs

```diff
@@ -35,6 +35,8 @@ use spicepod::component::secrets::SpiceSecretStore;
 pub enum Error {
     #[snafu(display("Unable to load secrets for {store}"))]
     UnableToLoadSecrets { store: String },
+    #[snafu(display("Unable to load secrets for {store}: {error}"))]
+    UnableToLoadSecretsWithError { store: String, error: String },
```

Comment:
This should follow the Snafu philosophy - the `source` field automatically gets captured with `.context` as long as the inner type of the error is put here.
`UnableToLoadSecretsWithError { store: String, source: <inner error type> },`

Reply:
Updated, thank you @phillipleblanc